lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -386,7 +386,16 @@ public function update($entity)
         $this->updateTable($entity, $quotedTableName, $data, $isVersioned);
 
         if ($isVersioned) {
-            $id = $this->em->getUnitOfWork()->getEntityIdentifier($entity);
+            $identifier = $this->em->getUnitOfWork()->getEntityIdentifier($entity);
+
+            $id = [];
+            foreach ($this->class->identifier as $idField) {
```

Comment:
This seems like a workaround for me... [as @Ocramius said](https://github.com/doctrine/doctrine2/issues/6393#issuecomment-293533668) we should rather use field names everywhere.

Reply:
This doesn't seem possible, because in `UnitOfWork::createEntity` it passes the whole $row ($data) into the identifier flattener and the hydration works by passing the association data as columns not as fields, because it must support composite primary keys as assocations where using field name as a single name would not be able to represent two ore more columns.

Reply:
ACtually the other way around, BasicEntityPersister::update used the wrong identifier representation. See #8384